src/log.c

```diff
@@ -429,22 +479,25 @@ void qd_vlog_impl(qd_log_source_t *source, qd_log_level_t level, bool check_leve
     DEQ_ITEM_INIT(entry);
 
     //
-    // Obtain the log_source_lock global lock. We need to do this, if not, the qd_log_entity() function
+    // Obtain the log_entries_lock. We need to do this, if not, the qd_log_entity() function
     // could free the log_source->sink from underneath you and replace it with a new sink.
     // Once we obtain this lock, we only release the lock once the log line is written to the sink.
     //
-    sys_mutex_lock(log_source_lock);
     entry->module = source->module ? strdup(source->module) : 0;
     entry->level  = level;
     entry->file   = file ? strdup(file) : 0;
     entry->line   = line;
     gettimeofday(&entry->time, NULL);
```

Comment:
We may need to move the gettimeofday into write_log so we're holding the lock while the timestamp is taken to reduce likelyhood of re-ordering log events.